data_filtering.py

```diff
@@ -0,0 +1,22 @@
+import numpy as np
+from scipy.signal import savgol_filter
+
+
+def filter_data_coordinates(data):
+    x_filter = np.array(data)[:, 0]
+    window_size = min(len(x_filter), 15)
+    window_size = window_size if window_size % 2 == 1 else window_size - 1
+    x_filter = savgol_filter(x_filter, window_size, min(window_size - 1, 2), mode='interp')
+
+    y_filter = np.array(data)[:, 1]
+    y_filter = savgol_filter(y_filter, window_size, min(window_size - 1, 2), mode='interp')
+    return x_filter, y_filter
+
+
+def filter_data_angle(data):
+    x_filter = np.array(data)
+    window_size = min(len(x_filter), 15)
+    window_size = window_size if window_size % 2 == 1 else window_size - 1
+    x_filter = savgol_filter(x_filter, window_size, min(window_size - 1, 2), mode='interp')
```

Comment:
aren't these functions the same?  can't you just define filter outside of function and call `filter_data_angle` twice to get the same data `fileter_data_coordinates` returns? If so, maybe consider renaming to `apply_savgol_filter(data, filter)` signature?